src/sinks/statsd.rs

```diff
@@ -282,10 +284,12 @@ impl Service<BytesMut> for StatsdSvc {
     type Error = crate::Error;
     type Future = future::BoxFuture<'static, Result<(), Self::Error>>;
 
+    // Emission of Error internal event is handled upstream by the caller
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
         self.inner.poll_ready(cx).map_err(Into::into)
     }
 
+    // Emission of Error internal event is handled upstream by the caller
```

Comment:
Nit.. these aren't really function docs. I'd put the comments into the function body.

Reply:
But highly approve of adding the comments!

Reply:
Hmmm. Would it make more sense to put them at the top of the `impl` (instead of having one each for the trait functions) ?

I ask partly because I've sprinkled this exact same comment into a dozen other places in the sinks (grep for "Emission of Error" and you'll see what I mean). In some of the other cases, the functions are actually doing stuff in the body.

Reply:
(I'm happy to go back and edit those other places for consistency too)

Reply:
It'd be superneat to be able to hover over the function and get the docs that the emit is handled or something 💭

Reply:
Yeah, a good alternative might be to make them proper doc comments with `///`.

Reply:
Humm. If I make them doc comments in the impls , it overrides the highlight helper for the trait definition. So, I don't quite like that.

I _could_ at it to the trait definition, and remove it from all the implementations. My only concern is that it might lead to developers not seeing that information and trying to emit internal events in the impls.

The other option aside from leaving as is , is to move the comment inside the function body.

I guess it is pretty common procedure to read the documentation for a trait function being implemented. So perhaps adding it to the trait function doc comment isn't a bad idea :thinking: 

I'm torn now :laughing:

Reply:
After sleeping on it. I like the approach of adding it to the Trait function definition, and it'll appear in the highlight hint for the function (tested this).